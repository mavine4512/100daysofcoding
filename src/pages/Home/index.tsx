import React from 'react';

function Home(): JSX.Element {
    return <div><h5>Welcome to react test system</h5></div>;
}

export default Home;
