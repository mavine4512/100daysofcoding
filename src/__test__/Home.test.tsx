import React from 'react';
import ReactDOM from 'react-dom';
import Home from "../pages/Home";
import {render} from "@testing-library/react";

it('renders correctly home', ()=>{
    const  div =document.createElement('h5');
    ReactDOM.render(<Home/>, div);
    ReactDOM.unmountComponentAtNode(div)
    expect(div).toMatchSnapshot();

})

it('renders Welcome message', ()=>{
    const {getByText,unmount} = render(<Home/>)
    expect(getByText('Welcome to react test system')).toBeInTheDocument()

    unmount()
})
