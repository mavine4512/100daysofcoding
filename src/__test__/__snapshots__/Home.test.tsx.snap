// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly home 1`] = `<h5 />`;
